scripts/reporting/ncanda_quality_control_script.py

```diff
@@ -0,0 +1,315 @@
+          'stroop_date','mrireport_missing','mrireport_date',
+          'mr_session_report_complete'];
+
+form_fields = [['youthreport1_missing','youthreport1_date'],
+          ['youthreport1b_missing', 'youthreport1b_date'],['youthreport2_missing',
+          'youthreport2_date'],['parentreport_missing','parentreport_date'],
+          ['ssage_youth_missing','ssage_youth_date'],['lssaga1_youth_missing',
+          'lssaga1_youth_date'],['lssaga1_parent_missing','lssaga1_parent_date'],
+          ['bio_np_missing', 'bio_np_date'],['dd1000_missing','dd1000_date'],
+          ['dd100_missing','dd100_date'],['np_wrat4_missing','np_wrat4_wr_raw'],
+          ['np_reyo_missing','np_reyo_copy_time'], ['np_atax_missing',
+          'np_atax_sht_trial1'],['np_wais4_missing', 'np_wais4_rawscore'],
+          ['pasat_missing','pasat_date'],['cnp_missing',
+          'cnp_test_sessions_dotest'],['stroop_missing','stroop_date']];
+
+np_gpeg_fields = [['np_gpeg_exclusion___dh','np_gpeg_dh_time'],
+                  ['np_gpeg_exclusion___ndh','np_gpeg_ndh_time']]
+
+saliva_fields = [['saliva_1_collected','saliva_1_date'],
+          ['saliva_2_collected','saliva_2_date'],['saliva_3_collected',
+          'saliva_3_date'],['saliva_4_collected','saliva_4_date']];
+
+fields_sex = [['youthreport1_missing','youthreport1_yid2'],
+			['youthreport2_missing','youthreport2_yid2']]
+
+def get_project_entry(args=None):
+    """
+    Pulls the data from REDCap
+    """
+    # Get API key.
+    summary_key_file = open(os.path.join(os.path.expanduser("~"),
+                                         '.server_config',
+                                         'redcap-dataentry-token'), 'r')
+    summary_api_key = summary_key_file.read().strip()
+
+    # Connect to API.
+    project_entry = redcap.Project('https://ncanda.sri.com/redcap/api/',
+                                   summary_api_key, verify_ssl=False)
+    return project_entry
+
+def data_entry_fields(fields,project,arm):
+    """
+    Gets the dataframe containing a specific arm from REDCap
+    """
+    # Get a dataframe of fields
+    data_entry_raw = project.export_records(fields=fields, format='df',
+        events=[arm])
+    return data_entry_raw
+
+def missing_form(idx,row,field_missing, field_value):
+    """
+    Generates a report indicating which Forms have not been entered onto redcap
+    """
+    error = dict()
+    #exclude with a value of 1 is excluded
+    if math.isnan(row.get('exclude')):
+        # visit_ignore____yes with value 1 is ignored
+        if row.get('visit_ignore___yes') != 1:
+            # form is not missing if form_missing if value nan or zero
+            if row.get(field_missing) != 1:
+                # for form_date, date is stored as a string
+                if type(row.get(field_value)) == float:
+                    if math.isnan(row.get(field_value)):
+                        error = dict(subject_site_id = idx[0],
```

Comment:
@abonil91 for errors lets define a new function for all the errors to repeat dulpication:

```
def error_log(id, error, **kwargs):
    error = dict(experiment_site_id_=id,
               error=error,
               **kwargs)
   return error
```